src/ops.rs

```diff
@@ -139,13 +139,24 @@ pub trait OpName {
     fn name(&self) -> SmolStr;
 }
 
+/// Trait statically querying the tag of an operation.
+///
+/// This is implemented by all OpType variants, and always contains the dynamic
+/// tag returned by `OpType::tag(&self)`.
+pub trait OpTagged {
+    /// The name of the operation.
+    const TAG: OpTag;
+}
+
 #[enum_dispatch]
 /// Trait implemented by all OpType variants.
 pub trait OpTrait {
     /// A human-readable description of the operation.
     fn description(&self) -> &str;
+
     /// Tag identifying the operation.
     fn tag(&self) -> OpTag;
```

Comment:
Can we make `OpTagged` a requirement for `OpTrait` so we don't have to keep writing the same logic for `tag`? Or even move `tag` into `OpTagged`

Reply:
`enum_dispatch` derives the `OpTrait` trait for `OpType` by calling each variant's method.
Implementing `OpTagged` for `OpType` could only return `Tag::Any`, which isn't that useful (and I'd say calling that method is probably a bug in the logic).

Would it be better to rename `OpTagged` to `StaticTag` or something ?

Reply:
Ah, maybe we can add a default implementation requiring `OpTagged`, so we avoid all the `<Self as OpTagged>::TAG` lines.